dask/array/creation.py

```diff
@@ -465,7 +465,15 @@ def eye(N, chunks, M=None, k=0, dtype=float):
 @wraps(np.diag)
 def diag(v):
     name = 'diag-' + tokenize(v)
-    if isinstance(v, np.ndarray):
+
+    if hasattr(v, '_meta'):
```

Comment:
Are there cases when we won't have this attribute?

Reply:
Ah, never mind, I see the `isinstance(v, np.ndarray)` type check in the old diff.

Reply:
Correct. :)